File: frontend/src/api/notification.ts
import { request } from '@utils/request'

export interface Notification {
  id: number
  userId: number
  type: string
  title: string
  content: string
  relatedType?: string
  relatedId?: number
  isRead: boolean
  createdAt: string
  readAt?: string
}

// 获取通知列表
export function getNotifications(params: { page: number; size: number; isRead?: boolean; type?: string }) {
  return request.get('/notifications', { params })
}

// 获取未读数量
export function getUnreadCount() {
  return request.get('/notifications/unread-count')
}

// 标记为已读
export function markAsRead(id: number) {
  return request.put(`/notifications/${id}/read`)
}

// 全部标记为已读
export function markAllAsRead() {
  return request.put('/notifications/read-all')
}

// 删除通知
export function deleteNotification(id: number) {
  return request.delete(`/notifications/${id}`)
}

File: backend/teamup-server/src/main/java/com/teamup/server/modules/notification/entity/Notification.java
package com.teamup.server.modules.notification.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 通知实体
 */
@Data
@TableName("notifications")
public class Notification {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 接收用户ID
     */
    private Long userId;
    
    /**
     * 通知标题
     */
    private String title;
    
    /**
     * 通知内容
     */
    private String content;
    
    /**
     * 通知类型
     */
    private String type;
    
    /**
     * 关联类型（如：TEAM, PROJECT, USER等）
     */
    private String relatedType;
    
    /**
     * 关联ID
     */
    private Long relatedId;
    
    /**
     * 是否已读
     */
    private Boolean isRead;
    
    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    /**
     * 阅读时间
     */
    private LocalDateTime readAt;
}
